### ISSUE TITLE
Error During A Upgrading - Angular2

### ISSUE THREAD

**Issue by vw98075** [jhipster/generator-jhipster#5811]
##### **Overview of the issue**

I can't upgrade to the 4.5.1 version due to an error shown in the followings:

~/jh/sampleentitymodel $ yo jhipster:upgrade
Welcome to the JHipster Upgrade Sub-Generator
This will upgrade your current application codebase to the latest JHipster version
Looking for latest generator-jhipster version...
New generator-jhipster version found: 4.5.1
Initialized a new git repository
Committed with message "Initial"
Created branch jhipster_upgrade
Cleaned up directory
Installing JHipster 4.3.0 locally
Installed generator-jhipster@4.3.0
Regenerating app with jhipster 4.3.0...
Successfully regenerated app with jhipster 4.3.0
Committed with message "Generated with JHipster 4.3.0"
Checked out branch "master"
Current code recorded as generated with version 4.3.0
Updating generator-jhipster to 4.5.1 . This might take some time...
Error jhipster:upgrade 

ERROR! Something went wrong while updating generator! yarn add v0.23.4
[1/4] Resolving packages...
[2/4] Fetching packages...
info "fsevents@1.1.1" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved 7 new dependencies.
├─ commander@2.9.0
├─ didyoumean@1.2.1
├─ generator-jhipster@4.5.1
├─ jhipster-core@1.3.2
├─ lodash.difference@4.5.0
├─ tabtab@2.2.2
└─ yeoman-environment@1.6.6
$ yarn run webpack:build
yarn run v0.23.4
$ yarn run ngc && webpack --config webpack/webpack.vendor.js && webpack --config webpack/webpack.dev.js 
yarn run v0.23.4
$ ngc -p tsconfig-aot.json 
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
 warning fsevents@1.1.1: The platform "linux" is incompatible with this module.
warning "angular2-cookie@1.2.6" has incorrect peer dependency "@angular/common@^2.0.0".
warning "angular2-cookie@1.2.6" has incorrect peer dependency "@angular/core@^2.0.0".
warning "ng2-translate@4.2.0" has incorrect peer dependency "@angular/core@^2.0.0".
warning "ng2-translate@4.2.0" has incorrect peer dependency "@angular/http@^2.0.0".
Unexpected value 'RegionMySuffixComponent in /home/oem/Workspace-dev/jh/sampleentitymodel/src/main/webapp/app/entities/region/index.ts' declared by the module 'SampleentitymodelRegionMySuffixModule in /home/oem/Workspace-dev/jh/sampleentitymodel/src/main/webapp/app/entities/region/region-my-suffix.module.ts'. Please add a @Pipe/@Directive/@Component annotation.
error Command failed with exit code 1.
error Command failed with exit code 1.
error Command failed with exit code 1.

##### **Motivation for or Use Case**

Upgrade the jHipster version

##### **JHipster Version(s)**

4.5.1 version of JHipster. It is a regression.
-->

##### **JHipster configuration**
Executing jhipster:info
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
sampleentitymodel@0.0.0 /home/oem/Workspace-dev/jh/sampleentitymodel
└── generator-jhipster@4.5.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

```yaml
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.vw.springsocialstudy"
    },
    "jhipsterVersion": "4.3.0",
    "baseName": "sampleentitymodel",
    "packageName": "com.vw.springsocialstudy",
    "packageFolder": "com/vw/springsocialstudy",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "no",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSocialSignIn": true,
    "rememberMeKey": "22f198d978340f45cb6bbdd5f89d66e18ddd1c20",
    "clientFramework": "angular2",
    "useSass": true,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

Country.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipType": "many-to-one",
            "relationshipName": "region",
            "otherEntityName": "region",
            "otherEntityField": "id"
        }
    ],
    "fields": [
        {
            "fieldName": "countryName",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170419205232",
    "entityTableName": "country",
    "dto": "no",
    "pagination": "no",
    "service": "serviceClass",
    "angularJSSuffix": "exc"
}
```

Department.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipType": "one-to-many",
            "javadoc": "A relationship",
            "relationshipName": "employee",
            "otherEntityName": "employee",
            "otherEntityRelationshipName": "department"
        },
        {
            "relationshipType": "many-to-one",
            "relationshipName": "location",
            "otherEntityName": "location",
            "otherEntityField": "id"
        }
    ],
    "fields": [
        {
            "fieldName": "departmentName",
            "fieldType": "String",
            "fieldValidateRules": [
                "required"
            ]
        }
    ],
    "changelogDate": "20170419205234",
    "entityTableName": "department",
    "dto": "no",
    "pagination": "no",
    "service": "serviceClass",
    "angularJSSuffix": "exc"
}
```

Employee.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "department",
            "otherEntityName": "department",
            "relationshipType": "many-to-one",
            "otherEntityField": "id"
        },
        {
            "relationshipType": "one-to-many",
            "relationshipName": "job",
            "otherEntityName": "job",
            "otherEntityRelationshipName": "employee"
        },
        {
            "relationshipType": "many-to-one",
            "relationshipName": "manager",
            "otherEntityName": "employee",
            "otherEntityField": "id"
        }
    ],
    "fields": [
        {
            "fieldName": "firstName",
            "javadoc": "The firstname attribute.",
            "fieldType": "String"
        },
        {
            "fieldName": "lastName",
            "fieldType": "String"
        },
        {
            "fieldName": "email",
            "fieldType": "String"
        },
        {
            "fieldName": "phoneNumber",
            "fieldType": "String"
        },
        {
            "fieldName": "hireDate",
            "fieldType": "ZonedDateTime"
        },
        {
            "fieldName": "salary",
            "fieldType": "Long"
        },
        {
            "fieldName": "commissionPct",
            "fieldType": "Long"
        }
    ],
    "changelogDate": "20170419205236",
    "javadoc": "The Employee entity.",
    "entityTableName": "employee",
    "dto": "no",
    "pagination": "infinite-scroll",
    "service": "no",
    "angularJSSuffix": "mySuffix"
}
```

Job.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "employee",
            "otherEntityName": "employee",
            "relationshipType": "many-to-one",
            "otherEntityField": "id"
        },
        {
            "relationshipType": "many-to-many",
            "otherEntityRelationshipName": "job",
            "relationshipName": "task",
            "otherEntityName": "task",
            "otherEntityField": "title",
            "ownerSide": true
        }
    ],
    "fields": [
        {
            "fieldName": "jobTitle",
            "fieldType": "String"
        },
        {
            "fieldName": "minSalary",
            "fieldType": "Long"
        },
        {
            "fieldName": "maxSalary",
            "fieldType": "Long"
        }
    ],
    "changelogDate": "20170419205237",
    "entityTableName": "job",
    "dto": "no",
    "pagination": "pagination",
    "service": "no",
    "angularJSSuffix": "mySuffix"
}
```

JobHistory.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipType": "many-to-one",
            "relationshipName": "job",
            "otherEntityName": "job",
            "otherEntityField": "id"
        },
        {
            "relationshipType": "many-to-one",
            "relationshipName": "department",
            "otherEntityName": "department",
            "otherEntityField": "id"
        },
        {
            "relationshipType": "many-to-one",
            "relationshipName": "employee",
            "otherEntityName": "employee",
            "otherEntityField": "id"
        }
    ],
    "fields": [
        {
            "fieldName": "startDate",
            "fieldType": "ZonedDateTime"
        },
        {
            "fieldName": "endDate",
            "fieldType": "ZonedDateTime"
        },
        {
            "fieldName": "language",
            "fieldType": "Language",
            "fieldValues": "FRENCH,ENGLISH,SPANISH"
        }
    ],
    "changelogDate": "20170419205238",
    "entityTableName": "job_history",
    "dto": "no",
    "pagination": "infinite-scroll",
    "service": "serviceClass",
    "angularJSSuffix": "exc"
}
```

Location.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipType": "many-to-one",
            "relationshipName": "country",
            "otherEntityName": "country",
            "otherEntityField": "id"
        }
    ],
    "fields": [
        {
            "fieldName": "streetAddress",
            "fieldType": "String"
        },
        {
            "fieldName": "postalCode",
            "fieldType": "String"
        },
        {
            "fieldName": "city",
            "fieldType": "String"
        },
        {
            "fieldName": "stateProvince",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170419205233",
    "javadoc": "not an ignored comment",
    "entityTableName": "location",
    "dto": "no",
    "pagination": "no",
    "service": "serviceClass",
    "angularJSSuffix": "exc"
}
```

Region.json

```yaml
{
    "fluentMethods": true,
    "relationships": [],
    "fields": [
        {
            "fieldName": "regionName",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170419205231",
    "entityTableName": "region",
    "dto": "no",
    "pagination": "no",
    "service": "serviceClass",
    "angularJSSuffix": "exc"
}
```

Task.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipType": "many-to-many",
            "relationshipName": "job",
            "otherEntityName": "job",
            "ownerSide": false,
            "otherEntityRelationshipName": "task"
        }
    ],
    "fields": [
        {
            "fieldName": "title",
            "fieldType": "String"
        },
        {
            "fieldName": "description",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170419205235",
    "javadoc": "Task entity.\n@author The JHipster team.",
    "entityTableName": "task",
    "dto": "no",
    "pagination": "no",
    "service": "serviceClass",
    "angularJSSuffix": "exc"
}
```

##### **Browsers and Operating System**

openjdk version "1.8.0_121"
OpenJDK Runtime Environment (build 1.8.0_121-8u121-b13-0ubuntu1.16.04.2-b13)
OpenJDK 64-Bit Server VM (build 25.121-b13, mixed mode)

git version 2.7.4

node: v6.10.0

npm: 4.5.0

bower: 1.8.0

gulp:
[12:52:27] CLI version 1.3.0

yeoman: 1.8.5

yarn: 0.23.4

Docker version 17.03.1-ce, build c6d412e

docker-compose version 1.8.0, build unknown

- [x ] Checking this box is mandatory (this is just to show you read everything)
em**

- [ x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
Seems like your webpack build is failing 
```
Unexpected value 'RegionMySuffixComponent in /home/oem/Workspace-dev/jh/sampleentitymodel/src/main/webapp/app/entities/region/index.ts' declared by the module 'SampleentitymodelRegionMySuffixModule in /home/oem/Workspace-dev/jh/sampleentitymodel/src/main/webapp/app/entities/region/region-my-suffix.module.ts'. Please add a @Pipe/@Directive/@component annotation.
```

did you do any customisation in the generated files?

**Comment by vw98075** [comment]
Hi, Deep,

The original version was done awhile ago. To my memory, I didn't do any changes to the front end code.

**Comment by jdubois** [comment]
Closed by #5883
